topics/single-cell/tutorials/pseudobulk-analysis/tutorial.md

```diff
@@ -0,0 +1,503 @@
+>
+> Importantly, ensure that the copying of this matrix as a raw counts layer is done carefully and correctly. To verify that your AnnData object contains the necessary raw counts layer, you can use the **[Inspect AnnData Object](https://toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0)** tool. This tool helps confirm the presence of raw counts and other essential metadata in your AnnData object before proceeding with pseudobulk analysis.
+>
+{: .tip}
+
+> <hands-on-title> Decoupler Pseudobulk </hands-on-title>
+>
+> 1. {% tool [Decoupler pseudo-bulk](toolshed.g2.bx.psu.edu/repos/ebi-gxa/decoupler_pseudobulk/decoupler_pseudobulk/1.4.0+galaxy8) %} tool with the following parameters:
+>     - {% icon param-file %} **Input AnnData file**: `AnnData for Pseudobulk` (Input dataset obtained > from Zenodo)
+>     - **Produce a list of genes to filter out per contrast?**: `No`
+>     - **Obs Fields to Merge**: *(Leave empty if not applicable)*
+>     - **Groupby column**: `cell_type` (Column containing cell type annotations)
+>     - **Sample Key column**: `individual` (Column containing individual sample identifiers)
+>     - **Layer**: `counts` (Layer containing raw gene expression counts)
+>     - **Factor Fields**: `disease` (Column in `adata.obs` specifying experimental factors. For edgeR, the first field should be the main contrast field, followed by  covariates.)
+>     - **Use Raw**: `No`
+>     - **Produce AnnData with Pseudo-bulk**: *(Optional, if yes, it will generate an h5ad output file with pseudobulks)*
+>     - **Minimum Cells**: `10`
+>     - **Produce plots**: `Yes`
+>     - **Minimum Counts**: `10`
+>     - **Minimum Total Counts**: `1000`
+>     - **Minimum Counts Per Gene Per Contrast Field**: `20` (Genes with fewer counts in specific contrasts are flagged in a separate file but not excluded from the results.)
+>     - **Enable Filtering by Expression**: `Yes`
+>     - **Plot Samples Figsize**: `13 13`
+>     - **Plot Filtering Figsize**: `13 13`
+>
+>    > <comment-title> Performing DEG within Clusters </comment-title>
+>    >
+>    > **Important!** The count matrix retrieved from this tool includes all of our samples `individual` aggregated by `cell_type`, which can be identified by the column headers. If you want to perform comparisons of conditions within clusters of each individual cell type, you will need to subset the relevant columns of the matrix and use them as your new count matrix. We will demonstrate this in the last section of this tutorial with detailed hands-on steps.
+>    {: .comment}
+>
+{: .hands_on}
+
+> <question-title>What are the outputs of the Decoupler tool?</question-title>
+>
+> 1. How many outputs does the Decoupler tool generate?
+> 2. What do the outputs represent? What is the interpretation of the plots?
+> 3. Which output(s) will we use in edgeR for differential expression analysis?
+>
+> > <solution-title>Solutions</solution-title>
```

Comment:
The solution boxes automatically get the prefix "Solution", so this title is a bit redundant. It is ok to have no title here if it doesn't make sense, then the box will just say "Solution"

```suggestion
> > <solution-title></solution-title>
```